### PR TITLE
8206311: Add docs-javase, docs-reference to CI build

### PR DESCRIPTION
**Pull request by erikj79**
Reposting this review as PR.

In order to regularly build the javase and reference docs images, I have made the following changes:

1. Introduced bundle targets for the javase and reference docs images.
2. To run this in CI without adding more unnecessary build time to the linux-x64 target (where the current docs are built), I created a new jib profile "docs". This will run in a separate task and will build all 3 docs targets.
3. To enable the docs targets to run standalone without having to build a full JDK, I had to tweak the "build-jdk" concept a bit, to allow it to also function when not cross compiling for certain parts of the build. Specifically the buildtools-modules can now be built by a "build-jdk" instead of the exploded image.
4. For symmetry, I renamed variables and targets involving docs images to always include one of "jdk", "javase" or "reference" in the name. See below for targets.

The new top level targets are:

docs-{jdk.javase,reference}-image:
Aliases for the existing docs-{jdk,javase,reference} added for consistency. The old docs-image is now an alias for docs-jdk-image.

all-docs-images:
Builds all 3 of the above docs images.

docs-{jdk,javase,reference}-bundles:
Builds the bundles for each kind of docs image. The existing docs-bundles is now an alias for docs-jdk-bundles.

all-docs-bundles:
Builds all the docs bundles.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8206311](https://bugs.openjdk.java.net/browse/JDK-8206311): Add docs-javase, docs-reference to CI build

### Reviewers
 * [Magnus Ihse Bursie](https://openjdk.java.net/census#ihse) (@magicus - **Reviewer**) ⚠️ Review applies to 197eac5c3f3e66ea4e983efdc2e929fd8f41ef49

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/99/head:pull/99`
`$ git checkout pull/99`

### CODE REVIEW

File: make/conf/jib-profiles.js [hunk 1]
```diff
@@ -680,20 +679,46 @@ var getJibProfilesProfiles = function (input, common, data) {
...
+        "docs": {
+            target_os: input.build_os,
+            target_cpu: input.build_cpu,
+            dependencies: [
+                "boot_jdk", "devkit", "graphviz", "pandoc", buildJdkDep,
+            ],
+            configure_args: [
+                "--enable-full-docs",
+                "--with-build-jdk=" + input.get(buildJdkDep, "home_path")
+                    + (input.build_os == "macosx" ? "/Contents/Home" : "")
```
Comment (magicus) [inline make/conf/jib-profiles.js:693]: I'm not fully understanding the platform selection mechanism here. Previously, docs were only built on linux-x64. I can see no such filtering in the new code; and this piece here even indicates assumed support for macosx.

Question: Is docs still built only on one platform? How is this achieved then?

**Review by magicus — COMMENTED** [review]
Looks good to me, but I'd like som clarification about the docs jib target.

File: make/conf/jib-profiles.js (hunk 1 above)
Comment (erikj79) [reply make/conf/jib-profiles.js:693]: I wanted to make the jib profile "docs" buildable on any platform (though I guess windows is likely not going to work well). This makes future development more flexible for local builds. The platform we choose to use for our official  CI/tier configurations is controlled elsewhere and will continue to be linux-x64.

**Review by magicus — APPROVED** [review]

**Pull request closed without merging**